### PR TITLE
NMS-16149: Added confd to Sentinel

### PR DESCRIPTION
**Pull request by christianpape**
JIRA: https://opennms.atlassian.net/browse/NMS-16149

### CODE REVIEW

**Review by RangerRick — APPROVED** [review]

**Review by indigo423 — APPROVED** [review]
@RangerRick @christianpape The minion-config-schema.yml was introduced just for the Minion portal, we read this schema file and use it dynamically to create a user interface in the cloud portal to allow users to configure a Minion. We don't have a portal to configure the Sentinel, I don't think we should introduce a sentinel-config-schema.yml we actually don't really need. What do you think?

**Comment by RangerRick** [comment]
@indigo423 ah, that's true, shouldn't be necessary

**Comment by mmahacek** [comment]
Instead of a read me file, the content should be added to the docs

**Comment by christianpape** [comment]
> @RangerRick @christianpape The minion-config-schema.yml was introduced just for the Minion portal, we read this schema file and use it dynamically to create a user interface cloud portal to allow users to configure a Minion. We don't have a portal to configure the Sentinel, I don't think we should introduce a sentinel-config-schema.yml we actually don't really need. What do you think?

Ok.

File: docs/modules/reference/pages/configuration/sentinel-features.adoc [hunk 1]
```diff
@@ -327,3 +327,81 @@ admin@sentinel> config:update
 
 Please note, that the file `/opt/sentinel/etc/junos-telemetry-interface.groovy` must also be provided manually,
 e.g. by manually copying it over from _{page-component-title}_.
+
+== Configuring Sentinel via confd
```
Comment (Bonrob2) [inline docs/modules/reference/pages/configuration/sentinel-features.adoc:331]:
```suggestion
== Configure Sentinel via confd
```

File: docs/modules/reference/pages/configuration/sentinel-features.adoc [hunk 2]
```diff
@@ -327,3 +327,81 @@ admin@sentinel> config:update
...
+
+=== Mounting
```
Comment (Bonrob2) [inline docs/modules/reference/pages/configuration/sentinel-features.adoc:333]:
```suggestion
=== Mount
```

File: docs/modules/reference/pages/configuration/sentinel-features.adoc [hunk 3]
```diff
@@ -327,3 +327,81 @@ admin@sentinel> config:update
...
+When starting the Sentinel container, mount a yaml file to the following path `/opt/sentinel/sentinel-config.yaml`.
```
Comment (Bonrob2) [inline docs/modules/reference/pages/configuration/sentinel-features.adoc:334]:
```suggestion
When starting the Sentinel container, mount a YAML file to the following path `/opt/sentinel/sentinel-config.yaml`.
```

File: docs/modules/reference/pages/configuration/sentinel-features.adoc [hunk 4]
```diff
@@ -327,3 +327,81 @@ admin@sentinel> config:update
...
+
+Any configuration provided to confd will overwrite configuration specified as environment variables. Direct overlay of
+specific configuration files will overwrite the corresponding config provided by confd.
```
Comment (Bonrob2) [inline docs/modules/reference/pages/configuration/sentinel-features.adoc:337]:
```suggestion

```

File: docs/modules/reference/pages/configuration/sentinel-features.adoc [hunk 5]
```diff
@@ -327,3 +327,81 @@ admin@sentinel> config:update
...
+broker-url: "<broker url>"
+id: "<id>"
+location: "<location>"
+----
+Config specified will be written to `etc/org.opennms.sentinel.controller.cfg`.
+
+==== User/Password
+Supplying the broker username/password via yaml file for configuration via confd is not supported.
+
+==== Sentinel Elasticsearch Config
```
Comment (Bonrob2) [inline docs/modules/reference/pages/configuration/sentinel-features.adoc:354]:
```suggestion
==== Sentinel Elasticsearch config
```

File: docs/modules/reference/pages/configuration/sentinel-features.adoc [hunk 6]
```diff
@@ -327,3 +327,81 @@ admin@sentinel> config:update
...
+----
+datasource:
+url: "jdbc:postgresql://localhost:5432/opennms"
+username: "postgres"
+password: "postgres"
+database-name: "opennms"
+----
+Config specified will be written to `etc/org.opennms.netmgt.distributed.datasource.cfg`.
+
+==== Sentinel Kafka Config
```
Comment (Bonrob2) [inline docs/modules/reference/pages/configuration/sentinel-features.adoc:377]:
```suggestion
==== Sentinel Kafka config
```

File: docs/modules/reference/pages/configuration/sentinel-features.adoc (hunk 5 above)
Comment (Bonrob2) [reply docs/modules/reference/pages/configuration/sentinel-features.adoc:354]: By the way, I don't like our heading capitalization rules, as we end up with things like this where most of the heading is capitalized and one word isn't. Sigh.

File: docs/modules/reference/pages/configuration/sentinel-features.adoc [hunk 7]
```diff
@@ -327,3 +327,81 @@ admin@sentinel> config:update
...
+ipc:
+kafka:
+bootstrap.servers: "my-kafka-ip-1:9092,my-kafka-ip-2:9092"
+group.id: "OpenNMS"
+----
+Config specified will be written to `etc/org.opennms.core.ipc.sink.kafka.cfg` and `etc/org.opennms.core.ipc.sink.kafka.consumer.cfg`.
+
+==== Telemetry Flow Adapters
+Individual flow adapters can be configured. See the example below for how to specify parameters and parsers. Any number
+of uniquely named listeners can be defined.
```
Comment (Bonrob2) [inline docs/modules/reference/pages/configuration/sentinel-features.adoc:389]:
```suggestion

```

File: docs/modules/reference/pages/configuration/sentinel-features.adoc [hunk 8]
```diff
@@ -327,3 +327,81 @@ admin@sentinel> config:update
...
+----
+ipc:
+kafka:
+bootstrap.servers: "my-kafka-ip-1:9092,my-kafka-ip-2:9092"
+group.id: "OpenNMS"
+----
+Config specified will be written to `etc/org.opennms.core.ipc.sink.kafka.cfg` and `etc/org.opennms.core.ipc.sink.kafka.consumer.cfg`.
+
+==== Telemetry Flow Adapters
+Individual flow adapters can be configured. See the example below for how to specify parameters and parsers. Any number
```
Comment (Bonrob2) [inline docs/modules/reference/pages/configuration/sentinel-features.adoc:388]:
```suggestion
You can configure individual flow adapters and any number of uniquely named listeners.
See the example below for how to specify parameters and parsers.
```

File: docs/modules/reference/pages/configuration/sentinel-features.adoc [hunk 9]
```diff
@@ -327,3 +327,81 @@ admin@sentinel> config:update
...
+flows:
+adapters:
+NetFlow-5:
+class-name: "org.opennms.netmgt.telemetry.protocols.netflow.adapter.netflow5.Netflow5Adapter"
+parameters:
+some-key: "some-value"
+----
+Config specified will be written to `deploy/confd-flows-feature.xml`.
+
+==== Instance Id
```
Comment (Bonrob2) [inline docs/modules/reference/pages/configuration/sentinel-features.adoc:402]:
```suggestion
==== Instance ID
```

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
Minor updates to the docs file. Thank you!

**Review by Bonrob2 — DISMISSED** [review]
docs LGTM. Thank you!

File: opennms-container/sentinel/.gitignore [hunk 10]
```diff
@@ -1 +1,2 @@
+sentinel-config-schema.yml
```
Comment (indigo423) [inline opennms-container/sentinel/.gitignore:1]: I think we can drop this one as well, -1 line 😁

**Review by indigo423 — DISMISSED** [review]
LGTM!

File: opennms-container/sentinel/.gitignore (hunk 10 above)
Comment (christianpape) [reply opennms-container/sentinel/.gitignore]: done

**Review by RangerRick — APPROVED** [review]

**Review by RangerRick — APPROVED** [review]

**Pull request merged**